cli/commands/genesis/root_test.go

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: BUSL-1.1
+//
+// Copyright (C) 2025, Berachain Foundation. All rights reserved.
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file of this repository and at www.mariadb.com/bsl11.
+//
+// ANY USE OF THE LICENSED WORK IN VIOLATION OF THIS LICENSE WILL AUTOMATICALLY
+// TERMINATE YOUR RIGHTS UNDER THIS LICENSE FOR THE CURRENT AND ALL OTHER
+// VERSIONS OF THE LICENSED WORK.
+//
+// THIS LICENSE DOES NOT GRANT YOU ANY RIGHT IN ANY TRADEMARK OR LOGO OF
+// LICENSOR OR ITS AFFILIATES (PROVIDED THAT YOU MAY USE A TRADEMARK OR LOGO OF
+// LICENSOR AS EXPRESSLY REQUIRED BY THIS LICENSE).
+//
+// TO THE EXTENT PERMITTED BY APPLICABLE LAW, THE LICENSED WORK IS PROVIDED ON
+// AN “AS IS” BASIS. LICENSOR HEREBY DISCLAIMS ALL WARRANTIES AND CONDITIONS,
+// EXPRESS OR IMPLIED, INCLUDING (WITHOUT LIMITATION) WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE, NON-INFRINGEMENT, AND
+// TITLE.
+
+package genesis_test
+
+import (
+	stdbytes "bytes"
+	"fmt"
+	"math/rand"
+	"reflect"
+	"testing"
+	"testing/quick"
+
+	"github.com/berachain/beacon-kit/cli/commands/genesis"
+	"github.com/berachain/beacon-kit/config/spec"
+	"github.com/berachain/beacon-kit/consensus-types/types"
+	"github.com/berachain/beacon-kit/primitives/common"
+	"github.com/berachain/beacon-kit/primitives/crypto"
+	"github.com/berachain/beacon-kit/primitives/math"
+	"github.com/berachain/beacon-kit/primitives/version"
+	statetransition "github.com/berachain/beacon-kit/testing/state-transition"
+	"github.com/stretchr/testify/require"
+)
+
+// generator for random deposits.
+type TestDeposits types.Deposits
+
+func (TestDeposits) Generate(rand *rand.Rand, size int) reflect.Value {
+	res := make(TestDeposits, size)
+	for i := range size {
+		var (
+			pubKey crypto.BLSPubkey
+			creds  types.WithdrawalCredentials
+			amount math.Gwei
+			sign   crypto.BLSSignature
+
+			err error
+		)
+		_, err = rand.Read(pubKey[:])
+		if err != nil {
+			panic(fmt.Errorf("failed generating random pubKey: %w", err))
+		}
+		_, err = rand.Read(creds[:])
+		if err != nil {
+			panic(fmt.Errorf("failed generating random cred: %w", err))
+		}
+		_, err = rand.Read(sign[:])
+		if err != nil {
+			panic(fmt.Errorf("failed generating random sign: %w", err))
+		}
+
+		res[i] = &types.Deposit{
+			Pubkey:      pubKey,
+			Credentials: creds,
+			Amount:      amount,
+			Signature:   sign,
+			Index:       0, // indexes will be set in order in the test
+		}
+	}
+	return reflect.ValueOf(res)
+}
+
+func TestCompareGenesisCmdWithStateProcessor(t *testing.T) {
```

Comment:
a property test comparing the two ways we have to generate the validator genesis root (CLI vs StateProcessor), to make sure they return the same output given the same inputs